subsys/bluetooth/host/hci_core.c

```diff
@@ -6095,24 +6975,96 @@ int bt_le_adv_start_internal(const struct bt_le_adv_param *param,
 				set_le_scan_enable(false);
 			}
 #endif /* defined(CONFIG_BT_OBSERVER) */
-			err = le_set_private_addr(param->id);
-			set_param.own_addr_type = BT_ADDR_LE_RANDOM;
+			err = le_adv_set_private_addr(adv);
+			*own_addr_type = BT_ADDR_LE_RANDOM;
 
 #if defined(CONFIG_BT_OBSERVER)
 			if (scan_enabled) {
 				set_le_scan_enable(true);
 			}
 #endif /* defined(CONFIG_BT_OBSERVER) */
+		} else {
+			err = le_adv_set_private_addr(adv);
+			*own_addr_type = BT_ADDR_LE_RANDOM;
 		}
 
 		if (err) {
 			return err;
 		}
+	}
 
+	return 0;
+}
+
+int bt_le_adv_start_legacy(const struct bt_le_adv_param *param,
+			   const struct bt_data *ad, size_t ad_len,
+			   const struct bt_data *sd, size_t sd_len,
+			   const bt_addr_le_t *peer)
+{
+	struct bt_hci_cp_le_set_adv_param set_param;
+	struct bt_conn *conn = NULL;
+	struct net_buf *buf;
+	bool dir_adv = (peer != NULL);
+	int err;
+	struct bt_le_ext_adv *adv;
+
+	if (!atomic_test_bit(bt_dev.flags, BT_DEV_READY)) {
+		return -EAGAIN;
+	}
+
+	if (!valid_adv_param(param, dir_adv)) {
+		return -EINVAL;
+	}
+
+	if (atomic_test_bit(bt_dev.flags, BT_DEV_ADVERTISING)) {
+		return -EALREADY;
+	}
+
+	if (!bt_le_adv_random_addr_check(param)) {
+		return -EINVAL;
+	}
+
+	(void)memset(&set_param, 0, sizeof(set_param));
+
+	set_param.min_interval = sys_cpu_to_le16(param->interval_min);
+	set_param.max_interval = sys_cpu_to_le16(param->interval_max);
+	set_param.channel_map  = 0x07;
```

Comment:
The old advertising API didn't have this option either.
There isn't much we can provide to aid the application to setting this value. So i would prefer to leave this out until we have some more infrastructure for determining performance of BT channels.

Reply:
Ok, it depends on how willing we are on changing the API in case someone would have the need and if we want to allow the application to control everything

Reply:
Adding it later is no problem. It can be added to bt_le_adv_param as a new parameter or to options as new options. Which is why I prefer to leave it out now, since this PR is already big :)

Reply:
Ok